### PR TITLE
feat: Add release workflow

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Add release workflow to the project.

Changes proposed in this pull request:

- Adds release workflow to the project
- Adds a document  describing steps to use the new release workflow

**Related issue(s)**
#134

### CODE REVIEW

File: .github/workflows/create-release.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+env:
+  IMAGE_REPO: europe-docker.pkg.dev/kyma-project/prod/runtime-watcher-skr
+  CODE_REPOSITORY: kyma-project/runtime-watcher
+on:
+  workflow_dispatch:
+    inputs:
+      name:
+        description: 'Existing tag for the new release'
+        default: ""
+        required: true
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply .github/workflows/create-release.yaml:12]: The reason I did not went into "full automation" is exactly what you mentioned: Waiting for the image to be built, error handling, etc.
If a new tag is created and prowjob fails to build the image, or builts it AFTER the timeout configured here, then what? We have to remove the tag and re-create it? This should also be described in "How to do the release". And so on. 
I found this way to be most robust. Yes, it's not fully automatic. But there's nothing that can go wrong, it's either success or failure and you don't have to "rollback" anything.
Let's discuss this further, I am not saying my approach is the best one - It just reflects my personal bias towards "foolproof" solutions, even if they require a bit manual effort.

File: .github/workflows/create-release.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Generate changelog
+        id: generate_changelog
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          LAST_RELEASE_TAG: ${{ needs.validate-release.outputs.last_release_tag }}
+        run: |
+          echo "Generating changelog from version: ${{ inputs.name }} to version: $LAST_RELEASE_TAG"
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply .github/workflows/create-release.yaml:71]: 👍

**Review by jeremyharisch — APPROVED** [review]

File: HOW_TO_RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Runtime Watcher Release Procedure
+
+### Steps
+
+1. Checkout the main branch
+2. Create a new tag for the release in the main branch and push it the repository. The tag should follow semantic versioning scheme, e.g: `1.2.3`
+   The new tag will trigger a prowjob that creates and publishes a Docker image with a runtime watcher executable.
+   The new tag will also trigger a release workflow.
```
Comment (mmitoraj) [inline HOW_TO_RELEASE.md:8]:
```suggestion
   The new tag will also trigger the release workflow.
```

File: HOW_TO_RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+Note 1: The release workflow will wait for a docker image to be published. If the image is not available within a configured time (15 minutes), the workflow will fail. In that case you need to manually re-run it once the docker image is ready. The expected image URL is: `europe-docker.pkg.dev/kyma-project/prod/runtime-watcher-skr:<NewTag>`
+
+Note 2: If you want to create a tag without triggering a release, just ensure the tag does not match a semantic versioning scheme.
```
Comment (mmitoraj) [inline HOW_TO_RELEASE.md:12]:
```suggestion
> **NOTE:** If you want to create a tag without triggering a release, just ensure the tag does not match the Semantic Versioning format.
```

**Review by mmitoraj — COMMENTED** [review]
I would suggest:
- moving the document to the **/docs** folder
- renaming the file so that it is written in lower case (`how_to_release.md`)
- introducing a link to the `how_to_release.md` file in the root `README.md` in the **Read more** section

**Review by jeremyharisch — APPROVED** [review]

**Review by jeremyharisch — APPROVED** [review]

**Pull request merged**